apps/api/src/app/workflows-v2/dto/workflow.dto.ts

```diff
@@ -0,0 +1,140 @@
+import { IsArray, IsBoolean, IsDefined, IsEnum, IsOptional, IsString, ValidateNested } from 'class-validator';
+import { Type } from 'class-transformer';
+
+import { JsonSchema } from '@novu/framework';
+import { ChannelTypeEnum, StepTypeEnum, WorkflowOriginEnum } from '@novu/shared';
+
+class Schema {
+  schema: JsonSchema;
+}
+export type SingleChannelPreference = {
+  defaultValue: boolean;
+  readOnly: boolean;
+};
+
+export type WorkflowPreferencesDto = {
+  defaultWorkflowPreference: SingleChannelPreference;
+  channels: {
+    [key in (typeof ChannelTypeEnum)[keyof typeof ChannelTypeEnum]]: SingleChannelPreference;
+  };
+};
+
+export type CreateWorkflowDto = Omit<WorkflowDto, '_id'>;
+
+export type UpdateWorkflowDto = Omit<WorkflowDto, '_id'>;
+
+export type WorkflowResponseDto = WorkflowDto & {
+  updatedAt: string;
+};
+
+export type ListWorkflowResponse = {
+  workflowSummaries: MinifiedResponseWorkflowDto[];
+  totalResults: number;
+};
+
+export type MinifiedResponseWorkflowDto = Pick<WorkflowResponseDto, 'workflowId' | 'tags' | 'updatedAt' | '_id'> & {
+  stepSummery: StepTypeEnum[];
+};
+
+export class StepDto {
+  @IsString()
+  @IsDefined()
+  code: string;
+
+  @IsString()
+  @IsDefined()
+  stepId: string;
+
+  @IsString()
+  @IsDefined()
+  type: StepTypeEnum;
+
+  @IsOptional()
+  @ValidateNested()
+  @Type(() => Schema)
+  controls?: Schema;
+
+  @IsOptional()
+  @ValidateNested()
+  @Type(() => ReplyCallbackDto)
+  replyCallback?: ReplyCallbackDto;
+
+  @IsOptional()
+  @IsBoolean()
+  active?: boolean;
+
+  @IsOptional()
+  @IsBoolean()
+  shouldStopOnFail?: boolean;
```

Comment:
We should remove all of these properties as they're not supported in Framework today.

Reply:
good call, i was not sure if we want to implement the missing part in the product in order to have a parity with v1. i guess we do not plan it in the moment

Reply:
Do you think it's worth keeping since we could support it natively in the UI?

Just thinking out loud:
- `active` essentially acts as a skip option an UI option controller.
- `shouldStopOnFail` is something we started adding to the framework, though it hasn't been fully implemented yet.
- `replyCallback` is a feature we have in v1, but not many users are currently utilizing it.